Fix `yarn outdated` missing values in URL column

devoto13 opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory. -->

**Summary**

<!-- Is the feature a substantial feature request? Please use https://github.com/yarnpkg/rfcs -->

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->

Adding failing test for a regression. See #3441 for more details.

**Test plan**

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->

bestander commented:
Thanks @devoto13.
I don't think a test is necessary, this is a straight forward integration issue.

Let's just fix is instead.
The fix for info command was in passing the right request header https://github.com/yarnpkg/yarn/blob/master/src/cli/commands/info.js#L63.

For outdated command the header needs to be passed via a couple of functions into checkOutdated https://github.com/yarnpkg/yarn/blob/master/src/registries/npm-registry.js#L89

devoto13 commented:
@bestander Apparently `checkOutdated` is used only by `yarn outdated` and `yarn upgrade-interactive` commands. Both commands need full response from registry to show `URL` column, so it was enough to just modify this method.

I've added correct header and removed test. PTAL

bestander commented:
Great work, @devoto13~
